server/src/main/java/org/opensearch/index/mapper/DocumentParser.java

```diff
@@ -561,6 +697,191 @@ private static void parseObjectOrField(ParseContext context, Mapper mapper) thro
         }
     }
 
+    /**
+     * Parses fields with dots as literal field names when disable_objects is enabled.
+     * This method treats dotted field names (e.g., "user.name", "metrics.cpu.usage") as flat fields
+     * rather than expanding them into nested object hierarchies.
+     * 
+     * @param context the parse context
+     * @param mapper the object mapper with disable_objects enabled
+     * @throws IOException if parsing fails
+     */
+    private static void parseFieldWithDots(ParseContext context, ObjectMapper mapper) throws IOException {
+        if (mapper.isEnabled() == false) {
+            context.parser().skipChildren();
+            return;
+        }
+
+        XContentParser parser = context.parser();
+        XContentParser.Token token = parser.currentToken();
+        
+        if (token == XContentParser.Token.VALUE_NULL) {
+            // the object is null, simply bail
+            return;
+        }
+
+        String currentFieldName = parser.currentName();
+        if (token.isValue()) {
+            throw new MapperParsingException(
+                "Document structure incompatibility: Field ["
+                    + currentFieldName
+                    + "] in object ["
+                    + mapper.name()
+                    + "] is defined as an object with disable_objects=true, but received a concrete value. "
+                    + "Expected format: an object containing flat dotted field names (e.g., {\"field.name\": \"value\"})."
+            );
+        }
+
+        // if we are at the end of the previous object, advance
+        if (token == XContentParser.Token.END_OBJECT) {
+            token = parser.nextToken();
+        }
+        if (token == XContentParser.Token.START_OBJECT) {
+            // if we are just starting an OBJECT, advance, this is the object we are parsing, we need the name first
+            token = parser.nextToken();
+        }
+
+        try {
+            assert token == XContentParser.Token.FIELD_NAME || token == XContentParser.Token.END_OBJECT;
+            context.incrementFieldCurrentDepth();
+            context.checkFieldDepthLimit();
+            
+            while (token != XContentParser.Token.END_OBJECT) {
```

Comment:
Is it possible to reuse existing methods for this ? 
Lets take what we can from existing flows such that originate from `parseObjectOrNested` - this has lot of duplicate code